server/pulp/plugins/file/distributor.py

```diff
@@ -212,9 +213,13 @@ def _symlink_unit(self, build_dir, unit, target_paths):
         :param target_paths: The list of paths the unit should be symlinked to.
         :type  target_paths: list of L{str}
         """
+        cwd = os.path.abspath(os.path.curdir)
         for target_path in target_paths:
-            # symlink_filename = os.path.join(build_dir, unit.unit_key['name'])
-            symlink_filename = os.path.join(build_dir, target_path)
+            # any two absolute paths always share at least the root element
```

Comment:
Wouldn't it be simpler to just ensure the target path is not absolute (leading /)?

Reply:
I'm wondering whether that wouldn't introduce backwards incompatibility within a bugfix; I wanted to make sure if there are users who legitimately used absolute or dotted path elements not to break them. But chances are I'm over complicating it.

Reply:
As said in https://pulp.plan.io/issues/3841#note-6, there is no valid use cases for absolute paths I know of.

Forcing absolute paths under `build_dir` at places depending on how much they happen to share with `cwd` does not make sense to me. Why should `/var/my_file` end up at `<build_dir>/my_file` just because it happens to share the `/var` with `cwd`?